instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/SnsImpl.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+// this class is only used from SnsAccess from method with @NoMuzzle annotation
+class SnsImpl {
+  static {
+    // Force loading of SnsClient; this ensures that an exception is thrown at this point when the
+    // SNS library is not present, which will cause SnsAccess to have enabled=false in library mode.
+    @SuppressWarnings("unused")
+    String ensureLoadedDummy = SnsClient.class.getName();
+  }
+
+  private SnsImpl() {}
+
+  static SdkRequest modifyRequest(
+      SdkRequest request, Context otelContext, TextMapPropagator messagingPropagator) {
+    if (messagingPropagator == null) {
+      return null;
+    } else if (request instanceof PublishRequest) {
+      return injectIntoPublishRequest((PublishRequest) request, otelContext, messagingPropagator);
+    } else {
+      // NB: We do not support PublishBatchRequest which was only introduced in 2.17.84.
+      // To add support, either raise the minimum requirement for the SNS-specific features, or use
+      // reflection + NoMuzzle (NB: Can use SnsRequest base class, MessageAttributeValue class,
+      // etc).
```

Comment:
```suggestion
      // NB: We do not support PublishBatchRequest which was only introduced in 2.17.84.
      // To add support, some targeted use of @NoMuzzle + checks that the needed class
      // is available should work.
      // See https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8830#discussion_r1247570985
```

Adapting the comment following https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8830/files#r1247915429